src/Parser/Mage/Frost/Modules/Features/BrainFreeze.js

```diff
@@ -145,61 +163,57 @@ class BrainFreeze extends Analyzer {
-						.icon(SPELLS.BRAIN_FREEZE.icon)
-						.actual(`${formatPercentage(this.overwrittenPercent)}% overwritten`)
-						.recommended(`Overwriting none is recommended`);
-				});
-		} else {
-			when(this.overwriteSuggestionThresholds)
-				.addSuggestion((suggest, actual, recommended) => {
-					return suggest(<React.Fragment>You overwrote {formatPercentage(this.overwrittenPercent)}% of your <SpellLink id={SPELLS.BRAIN_FREEZE.id} /> procs. Try to use your procs as soon as possible to avoid this.</React.Fragment>)
-						.icon(SPELLS.BRAIN_FREEZE.icon)
-						.actual(`${formatPercentage(this.overwrittenPercent)}% overwritten`)
-						.recommended(`Overwriting none is recommended`);
-				});
-		}
-
-		when(this.expiredSuggestionThresholds)
-			.addSuggestion((suggest, actual, recommended) => {
-				return suggest(<React.Fragment>You allowed {formatPercentage(this.expiredPercent)}% of your <SpellLink id={SPELLS.BRAIN_FREEZE.id} /> procs to expire. Try to use your procs as soon as possible to avoid this.</React.Fragment>)
-					.icon(SPELLS.BRAIN_FREEZE.icon)
-					.actual(`${formatPercentage(this.expiredPercent)}% expired`)
-					.recommended(`Letting none expire is recommended`);
-			});
-
-		when(this.flurryWithoutProcSuggestionThresholds)
-			.addSuggestion((suggest, actual, recommended) => {
-				return suggest(<React.Fragment>You cast <SpellLink id={SPELLS.FLURRY.id} /> without <SpellLink id={SPELLS.BRAIN_FREEZE.id} /> or <SpellLink id={SPELLS.WINTERS_REACH_TRAIT.id} /> {this.flurryWithoutProc} times. The only time it is acceptable to hard case Flurry is if you have a proc of the Winter's Reach Azerite Trait.</React.Fragment>)
-					.icon(SPELLS.FLURRY.icon)
-					.actual(`${formatNumber(this.flurryWithoutProc)} casts`)
-					.recommended(`Casting none is recommended`);
-			});
-	}
-
-	statistic() {
+  }
+
+  suggestions(when) {
+
+    when(this.overwriteSuggestionThresholds)
+      .addSuggestion((suggest, actual, recommended) => {
+        let suggestBuilder;
+        if (this.glacialSpikeTalented) {
+          suggestBuilder = suggest(<React.Fragment>You overwrote {formatPercentage(this.overwrittenPercent)}% of your <SpellLink id={SPELLS.BRAIN_FREEZE.id} /> procs incorrectly. You should hold your proc for <SpellLink id={SPELLS.GLACIAL_SPIKE_TALENT.id} /> if you have {ICICLE_BREAKPOINT} or more <SpellLink id={SPELLS.ICICLES_BUFF.id} /> during your <SpellLink id={SPELLS.FROSTBOLT.id} /> cast, otherwise you should use it immediately.</React.Fragment>);
```

Comment:
This suggestion isnt very clear, and it assumes they are already casting Frostbolt ... which they might not be.

I would suggest something along the lines of 

You should hold your proc for <SpellLink id={SPELLS.GLACIAL_SPIKE_TALENT.id} /> if you have {ICICLE_BREAKPOINT} or more <SpellLink id={SPELLS.ICICLES_BUFF.id} /> or if you will have {ICICLE_BREAKPOINT} icicles after your current cast completes, otherwise you should use it immediately.

Reply:
The reason it mentions Frostbolt is because you only cast brain freeze after a Frostbolt to shatter it. Otherwise it's a mistake to just cast the flurry and ice lance if it wasn't before the FB. So the thought process the player should have is "do I have 3 or fewer icicles as I'm casting this FB?" If so, they cast flurry right as the FB finishes, otherwise they hold the proc.

Reply:
I'm not sure if there is recommended max length on suggestions, but if there isn't I would be more than happy to explain more in-depth how BF holding works in the suggestion itself to try and make it clearer. EX:

You overwrote 5% of your Brain Freeze procs incorrectly. You should use Brain Freeze immediately following a Frostbolt if you have fewer than 3 Icicles during the cast. If you have 3 or more icicles during the Frostbolt cast, save the Brain Freeze proc to Shatter Glacial Spike.